CMakeLists.txt

```diff
@@ -27,29 +28,130 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
 set(CMAKE_POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
 
-if(FIND_PM_DEPS)
-  find_package(SwiftSystem CONFIG REQUIRED)
-  find_package(TSC CONFIG REQUIRED)
+# Toolchain Vended dependencies
+find_package(dispatch QUIET)
+find_package(Foundation QUIET)
+
+include(FetchContent)
+
+set(_SPM_SAVED_BUILD_TESTING ${BUILD_TESTING})
+set(_SPM_SAVED_BUILD_EXAMPLES ${BUILD_EXAMPLES})
+set(_SPM_VENDOR_DEPENDENCIES)
+
+set(BUILD_EXAMPLES NO)
+set(BUILD_TESTING NO)
+
+find_package(ArgumentParser CONFIG)
+if(NOT ArgumentParser_FOUND)
+  message("-- Vending swift-argument-parser")
+  FetchContent_Declare(ArgumentParser
+    GIT_REPOSITORY https://github.com/apple/swift-argument-parser
+    GIT_TAG 1.2.3)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ArgumentParser)
+endif()
+
+find_package(TSC CONFIG)
+if(NOT TSC_FOUND)
+  message("-- Vending swift-tools-support-core")
+  FetchContent_Declare(ToolsSupportCore
+    GIT_REPOSITORY https://github.com/apple/swift-tools-support-core
+    GIT_TAG main)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ToolsSupportCore)
+endif()
 
-  find_package(LLBuild CONFIG)
-  if(NOT LLBuild_FOUND)
+find_package(LLBuild CONFIG)
+if(NOT LLBuild_FOUND)
+  if(APPLE)
     find_package(LLBuild REQUIRED)
+  else()
+    message("-- Vending swift-llbuild")
+    set(LLBUILD_SUPPORT_BINDINGS Swift)
+    FetchContent_Declare(LLBuild
+      GIT_REPOSITORY https://github.com/apple/swift-llbuild
+      GIT_TAG main)
+    list(APPEND _SPM_VENDOR_DEPENDENCIES LLBuild)
   endif()
+endif()
 
-  find_package(ArgumentParser CONFIG REQUIRED)
-  find_package(SwiftDriver CONFIG REQUIRED)
-  find_package(SwiftCollections CONFIG REQUIRED)
-  find_package(SwiftASN1 CONFIG REQUIRED)
-  find_package(SwiftCertificates CONFIG REQUIRED)
-  find_package(SwiftCrypto CONFIG REQUIRED)
+find_package(SwiftASN1 CONFIG)
+if(NOT SwiftASN1_FOUND)
+  message("-- Vending swift-asn1")
+  FetchContent_Declare(ASN1
+    GIT_REPOSITORY https://github.com/apple/swift-asn1
+    GIT_TAG 1.1.0)
+  list(APPEND _SPM_VENDOR_DEPENDENCIES ASN1)
+endif()
+
+find_package(SwiftCertificates CONFIG)
+if(NOT SwiftCertificates_FOUND)
+  message("-- Vending swift-certificates")
+  FetchContent_Declare(Certificates
+    # GIT_REPOSITORY https://github.com/apple/swift-certificates
+    # GIT_TAG 1.1.0)
+    GIT_REPOSITORY https://github.com/compnerd/swift-certificates
+    GIT_TAG vendor)
```

Comment:
This should probably be updated?

Reply:
Ah, right that should get updated - need to get https://github.com/apple/swift-certificates/pull/158 merged for that. Wouldn't mind a bit of help there!

Reply:
Done